### DISCUSSION TITLE
Kedro use cases for Machine Learning (ML) on Databricks

### DISCUSSION THREAD

**Discussion by nthanhtung** [kedro-org/kedro#1038] [Idea]
Databricks has native features that support end to end ML : 
* Features: AutoML, Managed ML Flow (Model Tracking, Registry, Serving), Feature Store, Repos (Built-in CICD), integration with Delta Lakehouse features (data governance, ETL)
* Demo: https://youtu.be/zQEiwJqqeeA , https://youtu.be/K9DIKk_XFFM
* What will be the use cases for Kedro in this scenario?

-----
Btw, still waiting for answers in this topic: https://github.com/quantumblacklabs/kedro/discussions/1009

**Comment by datajoely** [comment]
Hi @nthanhtung - Kedro isn't a substitute for databricks it can fit in as a lightweight mechanism for building out ML pipelines in all of these platform solutions be it Databricks or Sagemaker or similar.

**Reply by nthanhtung** [reply]
Hi @datajoely , what is the best practice to leverage both Databricks native ML features and Kedro features?
In which use case of bulding ML pipelines we should:
* only use Kedro features
* only use Databricks native ML features
* use both Kedro & Databricks

> Databricks has native features that support end to end ML :
> 
> * Features: AutoML, Managed ML Flow (Model Tracking, Registry, Serving), Feature Store, Repos (Built-in CICD), integration with Delta Lakehouse features (data governance, ETL)
> * Demo: https://youtu.be/zQEiwJqqeeA , https://youtu.be/K9DIKk_XFFM

**Reply by datajoely** [reply]
It's the final point - I'd really encourage you to follow the [spaceflights tutorial](https://kedro.readthedocs.io/en/stable/03_tutorial/01_spaceflights_tutorial.html) here to better understand what Kedro is and isn't.

**Reply by nthanhtung** [reply]
@datajoely , I have installed Kedro & followed the above tutorial to create a workflow: transform, merge data, then do ML, view linage

But I can also do similar things using Databricks delta live table ETL (SQL transformation, linage, data quality, dependency), then auto ML

So I still haven't got an answer for my above questions
